### PR TITLE
Relaxing requirements for installing SCX and OMI packages under --force

### PR DESCRIPTION
**Pull request by lagalbra**
Due to other Azure VM Extensions using the packages we depend on (SCX, OMI), the installation of the omsagent has been failing when these packages cannot be upgraded.
When the OMSAgent is installed via Azure extension, `--upgrade --force`, and the goal is to do as much as possible to succeed installation.
This change still checks that any dependencies fulfill the minimum requirements for the omsagent package to be installed, but does not require any check more than that if `--force` is used.

@Microsoft/omsagent-devs

### CODE REVIEW

**Comment by lagalbra** [comment]
Unit and system tests have passed and pBuild is running now.

I have tested that upgrading, normal functions, and purging a bundle work correctly, both with and without `--force`.

File: installer/bundle/bundle_skel.sh [hunk 1]
```diff
@@ -96,64 +99,87 @@ cleanup_and_exit()
...
+    elif [ $A_BUILD -gt $B_BUILD ]; then
         return 1
     fi
 
-    # Version available is idential to installed version, so don't install
+    # VersionA is idential to VersionB
     return 1
 }
 
+check_version_depending_on_force() {
```
Comment (sugr4) [inline installer/bundle/bundle_skel.sh]: Looks good to me. But can we improve this function name? maybe use the old name check_version_installable, and add a comment each before lines 177 and 179? It's up to you @lagalbra

**Comment by lagalbra** [comment]
Should note: this PR is mainly for code review purposes. I am going to test this change in the Azure Linux VM Extension before merging this change to master.

File: installer/bundle/bundle_skel.sh (hunk 1 above)
Comment (lagalbra) [inline installer/bundle/bundle_skel.sh]: Will do. I had created this name before I renamed the original `check_version_installable` to `first_version_lower_than_second`

**Pull request closed without merging**